### PR TITLE
[CORE] Add a config to fall back all regexp expressions

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

1. Add a config to let user fall back all regexp expressions if incompatibility is not tolerant.
2. Create a new module test/ in gluten-ut to place some common tests, not specific to a backend or spark version.

## How was this patch tested?

Added test.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
@ulysses-you, could you review this pr? Thanks!

**Review by ulysses-you — APPROVED** [review]
lgtm

File: docs/velox-backend-limitations.md [hunk 1]
```diff
@@ -25,9 +25,15 @@ Velox BloomFilter's serialization format is different from Spark's. BloomFilter
...
-such as `rlike`, `regexp_extract`, etc.
+#### Regexp functions
+In Velox, regexp functions (`rlike`, `regexp_extract`, etc.) are implemented based on RE2, while in Spark they are based on `java.util.regex`.
+* Lookaround (lookahead/lookbehind) pattern is not supported in RE2.
+* When matching white space with pattern "\\s", RE2 doesn't treat "\v" (or "\x0b") as white space, but `java.util.regex` does.
+
+There are a few unknown incompatible cases. If user cannot tolerate the incompatibility risk, please enable the below configuration property.
+```
+spark.gluten.sql.fallbackRegexpExpressions
+```
```
Comment (FelixYBW) [inline docs/velox-backend-limitations.md:36]: @codyschierbeck Do you have a list of result mismatch cases between re2 and Spark?

**Comment by PHILO-HE** [comment]
Let's merge this pr firstly. Will include more known incompatible cases found by @codyschierbeck in separate PR. Thanks!

File: docs/velox-backend-limitations.md (hunk 1 above)
Comment (FelixYBW) [reply docs/velox-backend-limitations.md:36]: image
@codyschierbeck Is this the list of unsupported pattens in re2?

**Pull request merged**